### PR TITLE
Support Pytorch 1.1.x

### PR DESCRIPTION
**Pull request by besirkurtulmus**

### CODE REVIEW

File: templates/pytorch-1.1.x/src/__ALGO__.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            if {'matrix_a', 'matrix_b'} <= input_dict.keys():
+                self.A = convert(input_dict['matrix_a'])
+                self.B = convert(input_dict['matrix_b'])
+            else:
+                raise Exception("'matrix_a' and 'matrix_b' must be defined.")
+        else:
+            raise Exception('input must be a json object.')
+        if self.A.shape[-1] != self.B.shape[0]:
+            raise Exception('inner dimensions between A and B must be the same.\n A: {} B: {}'.format(self.A.shape[-1],
+                                                                                                      self.B.shape[0]))
```
Comment (jamesatha) [inline templates/pytorch-1.1.x/src/__ALGO__.py]: nit on formatting. Can we put this on the previous line? or maybe put the 2 args on this line (and indent a bit less

File: templates/pytorch-1.1.x/src/__ALGO__.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    gpu_tensor = th_tensor.cuda()
+    return gpu_tensor
+
+def apply(input):
+    """
+    Calculates the dot product of two matricies using pytorch, with a cudnn backend.
+    Returns the product as the output.
+    """
+    input = InputObject(input)
+    C = th.mm(input.A, input.B)
```
Comment (jamesatha) [inline templates/pytorch-1.1.x/src/__ALGO__.py]: Doesn't look like the `InputObject` is really needed to be an object

Comment (besirkurtulmus) [reply templates/pytorch-1.1.x/src/__ALGO__.py]: I copied this over from the previous pytorch 1.0.0 template.
I think this is a valid way of parsing inputs outside of the apply() function.

File: templates/pytorch-1.1.x/src/__ALGO__.py [hunk 3]
```diff
@@ -32,8 +32,8 @@ def __init__(self, input_dict):
         else:
             raise Exception('input must be a json object.')
         if self.A.shape[-1] != self.B.shape[0]:
-            raise Exception('inner dimensions between A and B must be the same.\n A: {} B: {}'.format(self.A.shape[-1],
-                                                                                                      self.B.shape[0]))
+            raise Exception('inner dimensions between A and B must be the same.\n A: {} B: {}'\
+                .format(self.A.shape[-1], self.B.shape[0]))
```
Comment (besirkurtulmus) [inline templates/pytorch-1.1.x/src/__ALGO__.py]: @jamesatha does this look better?

File: templates/pytorch-1.1.x/src/__ALGO__.py (hunk 1 above)
Comment (besirkurtulmus) [reply templates/pytorch-1.1.x/src/__ALGO__.py]: Fixed it.

**Comment by besirkurtulmus** [comment]
@zeryx Yes, I have confirmed it.

File: templates/pytorch-1.0.0/src/__ALGO__.py [hunk 4]
```diff
@@ -32,8 +32,8 @@ def __init__(self, input_dict):
         else:
             raise Exception('input must be a json object.')
         if self.A.shape[-1] != self.B.shape[0]:
-            raise Exception('inner dimensions between A and B must be the same.\n A: {} B: {}'.format(self.A.shape[-1],
-                                                                                                      self.B.shape[0]))
+            raise Exception('inner dimensions between A and B must be the same.\n A: {} B: {}'\
```
Comment (jamesatha) [inline templates/pytorch-1.0.0/src/__ALGO__.py]: nit. keep the `.format(` on the previous line so you don't need the slash

File: templates/pytorch-1.1.x/src/__ALGO__.py (hunk 3 above)
Comment (jamesatha) [reply templates/pytorch-1.1.x/src/__ALGO__.py]: I'm not a fan of the slash. I'd move the method to the previous line and have the args on the second line. But don't feel too strongly

File: templates/pytorch-1.1.x/src/__ALGO__.py (hunk 2 above)
Comment (jamesatha) [reply templates/pytorch-1.1.x/src/__ALGO__.py]: I guess for consistencies sake, we can keep this as is, but it seems like a method that does the verification of these 2 dimensions would be nicer. But not  going not going to block this PR for that

File: templates/pytorch-1.1.x/src/__ALGO___test.py [hunk 5]
```diff
@@ -0,0 +1,8 @@
+from __ALGO__ import apply
+
+def test_algorithm():
+    input = {"matrix_a": [[0, 1], [1, 0]], "matrix_b": [[25, 25], [11, 11]]}
+    result = apply(input)
+    assert result == {"product": [[11., 11.], [25., 25.]]}
```
Comment (jamesatha) [inline templates/pytorch-1.1.x/src/__ALGO___test.py]: do we really need the dots? 11 and `11.` are going to be equal always, right/

Comment (besirkurtulmus) [reply templates/pytorch-1.1.x/src/__ALGO___test.py]: I think the results are floats, hence why we need the dots.

Comment (besirkurtulmus) [reply templates/pytorch-1.1.x/src/__ALGO___test.py]: Also, I think we want to be explicit about it.

**Pull request merged**